staging/src/k8s.io/kubelet/config/v1/types.go

```diff
@@ -88,6 +88,42 @@ type CredentialProvider struct {
 	// to pass argument to the plugin.
 	// +optional
 	Env []ExecEnvVar `json:"env,omitempty"`
+
+	// TokenAttributes is the configuration for the service account token that will be passed to the plugin.
+	// The credential provider opts in to using service account tokens for image pull by setting this field.
+	// When this field is set, kubelet will generate a service account token bound to the pod for which the
+	// image is being pulled and pass to the plugin as part of CredentialProviderRequest along with other
+	// attributes required by the plugin.
+	//
+	// The service account metadata and token attributes will be used as a dimension to cache
+	// the credentials in kubelet. The cache key is generated by combining the service account metadata
+	// (namespace, name, UID, and annotations key+value for the keys defined in ServiceAccountTokenAttribute.serviceAccountAnnotationKeys).
+	// +optional
+	TokenAttributes *ServiceAccountTokenAttributes `json:"tokenAttributes,omitempty"`
+}
+
+// ServiceAccountTokenAttributes is the configuration for the service account token that will be passed to the plugin.
+type ServiceAccountTokenAttributes struct {
+	// ServiceAccountTokenAudience is the intended audience for the projected service account token.
+	// +required
+	ServiceAccountTokenAudience string `json:"serviceAccountTokenAudience"`
+
+	// AllowWithoutServiceAccount indicates whether the plugin wants to be invoked even if the pod does not have a service account.
+	// If set to true, kubelet will invoke the plugin even if the pod does not have a service account. This is useful for plugins that
+	// are used to pull images for pods without service accounts (e.g., static pods).
+	// If set to false, kubelet will not invoke the plugin if the pod does not have a service account.
+	// +required
+	AllowWithoutServiceAccount *bool `json:"allowWithoutServiceAccount"`
```

Comment:
A positive name like `RequireServiceAccount *bool` might be simpler to reason about, documenting both behaviors and making the field required as you did.

> If `true`, only pods with matching images and with service accounts get routed to this provider.
> 
> If `false`, pods with matching images but without service accounts (like static pods) will still get routed to this provider but will not include a token.

Reply:
i've updated the API to use `RequireServiceAccount`